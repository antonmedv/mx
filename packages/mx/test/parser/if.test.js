const test = require('ava')
const parse = require('./parse')

test('parse if/else', t => {
  const code = `
  if foo == "value"
    p Foo is value
  else if bar
    p Bar is ok
  else
    p Okay
  `
  t.snapshot(parse(code))
})
