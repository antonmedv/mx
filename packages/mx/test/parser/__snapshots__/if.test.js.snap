// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse if/else 1`] = `
DocumentNode {
  "children": Array [
    IfStatementNode {
      "cond": BinaryExpressionNode {
        "left": IdentifierNode {
          "loc": SourceLocation {
            "end": Position {
              "column": 6,
              "line": 1,
            },
            "source": "file.mx",
            "start": Position {
              "column": 3,
              "line": 1,
            },
          },
          "name": "foo",
          "type": "Identifier",
        },
        "loc": SourceLocation {
          "end": Position {
            "column": 17,
            "line": 1,
          },
          "source": "file.mx",
          "start": Position {
            "column": 3,
            "line": 1,
          },
        },
        "operator": "==",
        "right": LiteralNode {
          "loc": SourceLocation {
            "end": Position {
              "column": 17,
              "line": 1,
            },
            "source": "file.mx",
            "start": Position {
              "column": 10,
              "line": 1,
            },
          },
          "type": "Literal",
          "value": "\\"value\\"",
        },
        "type": "BinaryExpression",
      },
      "loc": SourceLocation {
        "end": Position {
          "column": 0,
          "line": 7,
        },
        "source": "file.mx",
        "start": Position {
          "column": 0,
          "line": 1,
        },
      },
      "otherwise": IfStatementNode {
        "cond": IdentifierNode {
          "loc": SourceLocation {
            "end": Position {
              "column": 11,
              "line": 3,
            },
            "source": "file.mx",
            "start": Position {
              "column": 8,
              "line": 3,
            },
          },
          "name": "bar",
          "type": "Identifier",
        },
        "loc": SourceLocation {
          "end": Position {
            "column": 0,
            "line": 7,
          },
          "source": "file.mx",
          "start": Position {
            "column": 5,
            "line": 3,
          },
        },
        "otherwise": Array [
          ElementNode {
            "attributes": Array [],
            "children": Array [
              TextNode {
                "loc": SourceLocation {
                  "end": Position {
                    "column": 8,
                    "line": 6,
                  },
                  "source": "file.mx",
                  "start": Position {
                    "column": 3,
                    "line": 6,
                  },
                },
                "text": " Okay",
                "type": "Text",
              },
            ],
            "loc": SourceLocation {
              "end": Position {
                "column": 0,
                "line": 7,
              },
              "source": "file.mx",
              "start": Position {
                "column": 2,
                "line": 6,
              },
            },
            "name": "p",
            "type": "Element",
          },
        ],
        "then": Array [
          ElementNode {
            "attributes": Array [],
            "children": Array [
              TextNode {
                "loc": SourceLocation {
                  "end": Position {
                    "column": 13,
                    "line": 4,
                  },
                  "source": "file.mx",
                  "start": Position {
                    "column": 3,
                    "line": 4,
                  },
                },
                "text": " Bar is ok",
                "type": "Text",
              },
            ],
            "loc": SourceLocation {
              "end": Position {
                "column": 0,
                "line": 5,
              },
              "source": "file.mx",
              "start": Position {
                "column": 2,
                "line": 4,
              },
            },
            "name": "p",
            "type": "Element",
          },
        ],
        "type": "IfStatement",
      },
      "then": Array [
        ElementNode {
          "attributes": Array [],
          "children": Array [
            TextNode {
              "loc": SourceLocation {
                "end": Position {
                  "column": 16,
                  "line": 2,
                },
                "source": "file.mx",
                "start": Position {
                  "column": 3,
                  "line": 2,
                },
              },
              "text": " Foo is value",
              "type": "Text",
            },
          ],
          "loc": SourceLocation {
            "end": Position {
              "column": 0,
              "line": 3,
            },
            "source": "file.mx",
            "start": Position {
              "column": 2,
              "line": 2,
            },
          },
          "name": "p",
          "type": "Element",
        },
      ],
      "type": "IfStatement",
    },
  ],
  "loc": SourceLocation {
    "end": Position {
      "column": 0,
      "line": 7,
    },
    "source": "file.mx",
    "start": Position {
      "column": 0,
      "line": 1,
    },
  },
  "type": "Document",
}
`;
